Add support of output editing in DEBUG node using JSONata

HiroyasuNishiyama opened the pull request:
<!--
## Before you hit that Submit button....

Please read our [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
before submitting a pull-request.

## Types of changes

What types of changes does your code introduce?
Put an `x` in the boxes that apply
-->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

<!--
If you want to raise a pull-request with a new feature, or a refactoring
of existing code, it **may well get rejected** if it hasn't been discussed on
the [forum](https://discourse.nodered.org) or
[slack team](https://nodered.org/slack) first.

-->

## Proposed changes

<!-- Describe the nature of this change. What problem does it address? -->
Currently, formatting output of debug node requires additional node in front of its input.
In many cases, it is handy for DEBUG node can create custom output for easily recognizing results.
This PR add a feature to allow DEBUG node to specify JSONata expression for custom output editing.

## Checklist
<!-- Put an `x` in the boxes that apply -->

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

knolleary commented:
Hi @HiroyasuNishiyama, thanks for this.

I would rather this was added as a third option to the existing 'output' field, rather than added as a whole new field.

<img width="464" alt="local_node-red" src="https://user-images.githubusercontent.com/51083/49518017-7fa2ee80-f895-11e8-8995-2a7fbcd05639.png">

Evaluating the expression introduces an asynchronous step in handling of the message - for 99% of users who don't want an expression, we should avoid introducing the overhead of an additional Promise in the code that does nothing.

dceejay commented:
performance of debug is already an issue under high volume updates - so we must ensure we don't add anything extra to processing paths if possible.

HiroyasuNishiyama commented:
Thank you for comments.
Updated code to use suggested settings UI for inputting JSONata expression.
Also fixed normal message processing path not to incur additional overhead.

knolleary commented on packages/node_modules/@node-red/nodes/locales/en-US/core/58-debug.html:
```diff
@@ -23,4 +23,5 @@ <h3>Details</h3>
        Clicking on the source node id will reveal that node within the workspace.</p>
     <p>The button on the node can be used to enable or disable its output. It is recommended to disable or remove any Debug nodes that are not being used.</p>
     <p>The node can also be configured to send all messages to the runtime log, or to send short (32 characters) to the status text under the debug node.</p>
+    <p>Output of this node can be edited using JSONata expression by specifying the expression on <b>Output</b> field.</p>
```
This line of text is no longer matches the latest changes and should be removed.

I would suggest changing the first line of help to:

```
<p>Displays selected message properties in the debug sidebar tab and optionally the runtime log. By default it displays <code>msg.payload</code>, but can be configured to display any property, the full message or the result of a JSONata expression.</p>
```

HiroyasuNishiyama commented:
Updated info texts according to your suggestion.
